### DISCUSSION TITLE
Unified documentation

### DISCUSSION THREAD

**Discussion by luizcmarin** [yiisoft/docs#212] [Ideas]
My idea is that all documentation from all repositories and all languages are in one place, 'docs', that is, all unified documentation.

In several places Yii already does this, for example:
- file CONTRIBUTING.md
```
   - [Yii goal and values](https://github.com/yiisoft/docs/blob/master/001-yii-values.md)
   - [Namespaces](https://github.com/yiisoft/docs/blob/master/004-namespaces.md)
   - [Git commit messages](https://github.com/yiisoft/docs/blob/master/006-git-commit-messages.md)
   - [Exceptions](https://github.com/yiisoft/docs/blob/master/007-exceptions.md)
   - [Interfaces](https://github.com/yiisoft/docs/blob/master/008-interfaces.md)
```

- active-record->docs->internals.md package:
```
This package can be tested globally or individually for each DBMS.
  - [MSSQL](https://github.com/yiisoft/db-mssql)
  - [MySQL/MariaDB](https://github.com/yiisoft/db-mysql)
  - [Oracle](https://github.com/yiisoft/db-oracle)
  - [PostgreSQL](https://github.com/yiisoft/db-pgsql)
  - [SQLite](https://github.com/yiisoft/db-sqlite)
```

Look at what it is like today:
```
   - Configure [storing request](https://github.com/yiisoft/request-provider?tab=readme-ov-file#general-usage).
   - Configure [parameters resolver](docs/guide/en/parameters-resolvers.md).
   - Create DTO ([request input](docs/guide/en/request-input.md)).
```

But now, look at the hydrator-validator package:
```
> The package provides a [hydrator](https://github.com/yiisoft/hydrator)
that also does [validation](https://github.com/yiisoft/validator), including raw data.
> It's useful when input data comes from a user, and you need to validate it and then put it into DTOs.
```

### What if everything looked like this:

```
> The package provides a [hydrator](https://github.com/yiisoft/docs/guide/en/package-name/hydrator)
that also does [validation](https://github.com/yiisoft/docs/guide/en/package-name/validator), including raw data.

   - Configure [parameters resolver](https://github.com/yiisoft/docs/guide/en/package-name/parameters-resolvers.md).
   - Create DTO ([request input](https://github.com/yiisoft/docs/guide/en/package-name/request-input.md)).

This package can be tested globally or individually for each DBMS.
  - [MSSQL](https://github.com/yiisoft/docs/guide/en/package-name/db-mssql)
  - [MySQL/MariaDB](https://github.com/yiisoft/docs/guide/en/package-name/db-mysql)
  - [Oracle](https://github.com/yiisoft/docs/guide/en/package-name/db-oracle)
  - [PostgreSQL](https://github.com/yiisoft/docs/guide/en/package-name/db-pgsql)
  - [SQLite](https://github.com/yiisoft/docs/guide/en/package-name/db-sqlite)
```

**What looks best in organization, MAINTENANCE and _RESEARCH_?**

Why not do this with all the documentation for all packages? 
Yii3 deserves quality documentation.

See also this resource https://docs.github.com/en/communities/documenting-your-project-with-wikis/about-wikis

**Comment by vjik** [comment]
Package documentation should be near package code. Ideally, in one repository. Benefits:

- easier to keep the documentation up to date;
- correct documentation for different package versions.

**Comment by luizcmarin** [comment]
5 benefits of document centralization

- Reduction of communication errors;
- Improved collaboration between teams;
- Ease of tracking changes;
- Maintaining an organized history of documents.
- End of version conflicts.

Centralizing documents facilitates collaboration between different teams involved in a project. Everyone has access to updated documents, being able to share ideas, make notes, add comments and make changes if necessary. 

This encourages more efficient collaboration, avoiding duplication of efforts and ensuring that everyone is aligned regarding project tasks and objectives: with the most up-to-date version in their hands!

**Comment by samdark** [comment]
Since Yii has independent packages, it makes sense to keep docs for these in separate repositories, as @vjik mentioned. We can have common docs but these should not replace package docs for sure.
